src/Apps/Order/Routes/Shipping2/Components/SavedAddresses2.tsx

```diff
@@ -1,148 +1,177 @@
 
 import {
+  FulfillmentValues,
   SavedAddressType,
-  ShippingAddressFormValues,
-  addressWithFallbackValues,
+  getAddressByID,
   getDefaultUserAddress,
 } from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
 import { useShippingContext } from "Apps/Order/Routes/Shipping2/Hooks/useShippingContext"
 import { useOrderTracking } from "Apps/Order/Hooks/useOrderTracking"
-
-export const NEW_ADDRESS = "NEW_ADDRESS"
-const PAGE_SIZE = 30
+import { useFormikContext } from "formik"
+import { extractNodes } from "Utils/extractNodes"
+import { compact } from "lodash"
+import { graphql, useFragment } from "react-relay"
+// import { SavedAddresses2_addressConnection$key } from "__generated__/SavedAddresses2_addressConnection.graphql"
+import { SavedAddresses2_me$key } from "__generated__/SavedAddresses2_me.graphql"
 
 export interface SavedAddressesProps {
-  relay: RelayRefetchProp
-  me: SavedAddresses2_me$data
   active: boolean
-  onSelect: (address: ShippingAddressFormValues) => void
+  me: SavedAddresses2_me$key | null
+  onSelect: (address: SavedAddressType) => void
 }
 
-const SavedAddresses: React.FC<SavedAddressesProps> = props => {
+export const SavedAddresses2: React.FC<SavedAddressesProps> = props => {
   const logger = createLogger("SavedAddresses.tsx")
-  const [activeModal, setActiveModal] = useState<AddressModalAction | null>(
-    null
-  )
+
   const shippingContext = useShippingContext()
+  const orderTracking = useOrderTracking()
+  const formikContext = useFormikContext<FulfillmentValues>()
 
-  const { onSelect, me, relay } = props
+  const [
+    addressModalAction,
+    setAddressModalAction,
+  ] = useState<AddressModalAction | null>(null)
 
-  const addressList = compact<SavedAddressType>(
-    extractNodes(me?.addressConnection) ?? []
+  const data = useFragment(
+    graphql`
+      fragment SavedAddresses2_me on Me
+        @argumentDefinitions(
+          first: { type: "Int", defaultValue: 30 }
+          last: { type: "Int" }
+          after: { type: "String" }
+          before: { type: "String" }
+        ) {
+        addressConnection(
+          first: $first
+          last: $last
+          before: $before
+          after: $after
+        ) {
+          edges {
+            node {
+              internalID
+              name
+              addressLine1
+              addressLine2
+              city
+              region
+              postalCode
+              country
+              phoneNumber
+              isDefault
+            }
+          }
+        }
+      }
+    `,
+    props.me
   )
 
-  const selectedSavedAddressId =
-    shippingContext.orderData.savedFulfillmentDetails?.selectedSavedAddressId
-  const [selectedAddressID, setSelectedAddressID] = useState<
-    string | undefined
-  >(
-    getBestAvailableAddress(
-      addressList,
-      selectedSavedAddressId,
-      shippingContext.orderData.availableShippingCountries
-    )?.internalID
+  const addressList = compact<SavedAddressType>(
```

Comment:
Todo: Added addressList to shippingContext because it was used in hooks. Should it be used here too? I think the issue was that SavedAddresses has its own special fragment to pull this data from.